expr/seriesaggregator_test.go

```diff
@@ -102,12 +94,16 @@ func testSeriesAggregate(name, agg string, in []models.Series, out []schema.Poin
 	if len(got) != len(out) {
 		t.Fatalf("case %q (%q): len output expected %d, got %d", name, agg, len(out), len(got))
 	}
+
+	// Use EPSILON to avoid floating-point precision errors
+	EPSILON := math.Nextafter(1, 2) - 1
+
 	for j, p := range got {
 		bothNaN := math.IsNaN(p.Val) && math.IsNaN(out[j].Val)
-		if (bothNaN || p.Val == out[j].Val) && p.Ts == out[j].Ts {
+		if (bothNaN || p.Val == out[j].Val || math.Abs(p.Val-out[j].Val) < EPSILON) && p.Ts == out[j].Ts {
```

Comment:
interesting. have you seen falsely triggering test failures due to rounding errors?

Reply:
Ah, I was, but the issue was in my math for the test data. I never went back and removed this.

Reply:
my solution to that problem has been to pick different test values that don't trigger this problem :p